### PR TITLE
Add migration `Migration::class`.

### PR DESCRIPTION
**Pull request by terabytesoftw**
| Q             | A
| ------------- | ---
| Is bugfix?    | ✔️
| New feature?  | ❌
| Breaks BC?    | ❌

### CODE REVIEW

**Review by darkdef — APPROVED** [review]
Split please. One featur on the one PR

**Comment by samdark** [comment]
I'd apply same priciple as in comments for https://github.com/yiisoft/log-target-db/pull/43

File: src/DbHelper.php [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+        return true;
+    }
+
+    /**
+     * Loads the fixture into the database.
+     *
+     * @throws Exception
+     * @throws InvalidConfigException
+     */
+    public static function createMigrationFromSqlDump(ConnectionInterface $db, string $fixture): void
```
Comment (samdark) [inline src/DbHelper.php:53]: We can have it in tests if needed but it's not usable for end user.

File: src/DbHelper.php [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+use Yiisoft\Db\Exception\Exception;
+use Yiisoft\Db\Exception\InvalidConfigException;
+
+use function explode;
+use function file_get_contents;
+use function trim;
+
+final class DbHelper
+{
+    public static function createMigration(DbCache $dbCache, bool $force = false): bool
```
Comment (samdark) [inline src/DbHelper.php:17]:
```suggestion
    public static function ensureTable(DbCache $dbCache): bool
```

I'd add `public static function `dropTable(DbCache $dbCache): bool` and remove `$force` option.

Comment (samdark) [reply src/DbHelper.php:17]: Also we need a table name/prefix there as parameter.

File: src/DbHelper.php [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    }
+
+    public static function dropTable(ConnectionInterface $db, string $table, bool $force = false): void
+    {
+        $command = $db->createCommand();
+
+        if ($db->getTableSchema($table, $force) !== null) {
+            $command->dropTable($table)->execute();
+        }
+    }
```
Comment (samdark) [inline src/DbHelper.php:40]:
```suggestion
    public static function dropTable(ConnectionInterface $db, string $table): void
    {
        $command = $db->createCommand();

        if ($db->getTableSchema($table, true) !== null) {
            $command->dropTable($table)->execute();
        }
    }
```

File: src/Migration/schema-mssql.sql [hunk 4]
```diff
@@ -0,0 +1,8 @@
+IF OBJECT_ID('[dbo].[cache]', 'U') IS NOT NULL DROP TABLE [dbo].[cache];
```
Comment (samdark) [inline src/Migration/schema-mssql.sql:1]: Do we need these in `src` if we have `DbHelper`?

File: src/DbCache.php [hunk 5]
```diff
@@ -29,9 +29,7 @@
 /**
  * DbCache stores cache data in a database table.
  *
- * Database schema could be initialized by applying migration:
- *
- * {@see \Yiisoft\Cache\Db\Migration\M202101140204CreateCache}.
+ * Database schema can be initialized, you can use the {@see Migration::ensureTable()} method for this.
```
Comment (samdark) [inline src/DbCache.php:32]:
```suggestion
 * Use {@see Migration::ensureTable()} to initialize database schema.
```

File: tests/Driver/Mssql/DbHelperTest.php [hunk 6]
```diff
@@ -4,23 +4,27 @@
...
+use Yiisoft\Db\Exception\Exception;
+use Yiisoft\Db\Exception\InvalidConfigException;
 
 /**
  * @group Mssql
  *
  * @psalm-suppress PropertyNotSetInConstructor
  */
-final class MigrationMssqlTest extends MigrationTest
+final class DbHelperTest extends AbstractDbHelperTest
```
Comment (samdark) [inline tests/Driver/Mssql/DbHelperTest.php:17]: There's no "DB helper" anymore so I'd rename it to `MigrationTest`.

File: tests/Driver/Mssql/DbCacheTest.php [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+        // set table prefix
+        $this->db->setTablePrefix('mssql_');
+
+        // create migration
+        Migration::ensureTable($this->db);
+
+        parent::setUp();
+    }
+
+    public function testPrefixTable(): void
```
Comment (samdark) [inline tests/Driver/Mssql/DbCacheTest.php:38]: That looks like testing migration, not the cache itself. I guess it should be done in migration test only.

**Review by samdark — COMMENTED** [review]
Code in `src` is 👍

Tests are messy a bit:

1. There are no "helpers" anymore but helper tests exists.
2. There are tests for migration specific functionality in DbCacheTest.
3. Migration functionality tests are in DbHelper and Migration tests... that's complicated.

File: tests/Driver/Mssql/DbCacheTest.php (hunk 7 above)
Comment (samdark) [reply tests/Driver/Mssql/DbCacheTest.php:38]: OK. That's valid.

File: tests/Support/MssqlHelper.php [hunk 8]
```diff
@@ -7,19 +7,17 @@
 use Yiisoft\Db\Connection\ConnectionInterface;
 use Yiisoft\Db\Mssql\Connection;
 use Yiisoft\Db\Mssql\Driver;
+use Yiisoft\Db\Mssql\Dsn;
 
 final class MssqlHelper extends ConnectionHelper
```
Comment (samdark) [inline tests/Support/MssqlHelper.php:12]: It is `MssqlConnectionFactory`, not helper.

**Review by samdark — APPROVED** [review]

**Pull request merged**